### ISSUE TITLE
Hide Blocked Communities from Search Results

### ISSUE THREAD

**Issue by 8ullyMaguire** [LemmyNet/lemmy#3595]
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Is your proposal related to a problem?
Currently, when a user blocks a community on LemmyNet, the blocked community still appears in search results. This can be problematic for users who want to completely avoid any content from blocked communities. 

### Describe the solution you'd like.
The solution I would like to propose is to implement a feature that allows users to hide blocked communities from search results by default. This means that when a user blocks a community, it will no longer appear in search results unless explicitly unblocked. This would provide a more personalized and tailored experience for users who want to avoid content from blocked communities altogether.

### Describe alternatives you've considered.
An alternative solution would be to keep the current behavior where blocked communities still appear in search results, but provide a user setting that allows users to toggle whether they want to see content from blocked communities in search results or not. This would give users the flexibility to choose their preferred behavior.

### Additional context.
By implementing this feature, Lemmy can enhance user control and customization, allowing users to curate their own experience on the platform. It would provide a more seamless and personalized browsing experience for users who have blocked communities and want to ensure they are not exposed to any content from those communities in search results.

**Comment by dessalines** [comment]
This is already done: https://github.com/LemmyNet/lemmy/blob/main/crates/db_views_actor/src/community_view.rs#L167
